### DISCUSSION TITLE
Segment Diamonds Not Transparent (how to fix?)

### DISCUSSION THREAD

**Discussion by GLMeece** [JanDeDobbeleer/oh-my-posh#2124] [Troubleshoot]
I've hacked away at my version of jandedebbeleer's theme. I've gotten the prompt pretty much how I prefer, _except_ I can't seem to get _some_ of the segment arrows to be transparent. They're either black or white, but not all segments show this anomaly.

Here's a clip of my prompt (some segment details blurred for confidentiality):
![oh-my-posh_segments_with_incorrect_diamond_colors](https://user-images.githubusercontent.com/7154245/164516964-02acebc7-dee1-4175-ab1e-914b49d02a40.jpg)

* You can see that right segment arrow for Path is black, whereas I would prefer it to be that segments background color.
* Same for the Git segment
* Note that both the Node (at least its right arrow) _and_ Python segments are as I'd prefer

**TL;DR**

How do I force the segment arrows to use that segment's background color?

**Comment by JanDeDobbeleer** [comment]
Can you share the config? It's tricky to see what the issue is otherwise.

**Reply by GLMeece** [reply]
Argggh! I've tried several different methods to get the theme posted here, but my company recently instituted a very draconian policy to block copying text and pasting it into a browser. I'll see what else I can do...

**Reply by JanDeDobbeleer** [reply]
Take a screenshot. I can copy paste text from png's on macOS 😄

**Comment by SirenGlitch** [comment]
Finally found someone else with the same issue! Full config is below:

```toml
console_title_template = '{{ .Shell }} in {{ .Folder }}'
version = 3
final_space = true

[palette]
  black = '#262B44'
  blue = '#4B95E9'
  green = '#59C9A5'
  lime = '#00ff00'
  orange = '#F07623'
  purple = '#8800dd'
  red = '#D81E5B'
  white = '#E0DEF4'
  yellow = '#F3AE35'

[secondary_prompt]
  template = '❯❯ '
  foreground = 'p:green'

[transient_prompt]
  template = '❯ '
  foreground = 'p:green'

[[blocks]]
  type = 'prompt'
  alignment = 'left'
  newline = true

  [[blocks.segments]]
    leading_diamond = ''
    trailing_diamond = ''
    template = ' {{ path .Path .Location }} '
    foreground = 'p:white'
    background = 'p:orange'
    type = 'path'
    style = 'diamond'

    [blocks.segments.properties]
      cache_duration = 'none'
      style = 'agnoster'

      [blocks.segments.properties.mapped_locations]
        '~/.config/*' = '.config'
        '~/dotfiles/.config/*' = 'd/.config'

  [[blocks.segments]]
    template = ' {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }} {{ end }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }}  {{ .Working.String }}{{ end }}{{ if .Staging.Changed }}  {{ .Staging.String }}{{ end }} '
    foreground = 'p:black'
    powerline_symbol = ''
    background = 'p:green'
    type = 'git'
    style = 'powerline'
    foreground_templates = ['{{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}', '{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}', '{{ if gt .Ahead 0 }}p:white{{ end }}']
    background_templates = ['{{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}', '{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}', '{{ if gt .Ahead 0 }}#49416D{{ end }}', '{{ if gt .Behind 0 }}#7A306C{{ end }}']

    [blocks.segments.properties]
      branch_max_length = 25
      cache_duration = 'none'
      fetch_status = true
      fetch_upstream_icon = true

  [[blocks.segments]]
    template = '  '
    foreground = 'p:white'
    powerline_symbol = ''
    background = 'p:yellow'
    type = 'root'
    style = 'powerline'

    [blocks.segments.properties]
      cache_duration = 'none'

  [[blocks.segments]]
    leading_diamond = '<transparent,background></>'
    trailing_diamond = ''
    template = ' {{ if gt .Code 0 }}{{ else }}{{ end }} '
    foreground = 'p:white'
    background = 'p:blue'
    type = 'status'
    style = 'diamond'
    background_templates = ['{{ if gt .Code 0 }}p:red{{ end }}']

    [blocks.segments.properties]
      always_enabled = true
      cache_duration = 'none'

[[blocks]]
  type = 'prompt'
  alignment = 'left'
  newline = true

  [[blocks.segments]]
    template = '❯'
    foreground = 'p:green'
    type = 'text'
    style = 'plain'

    [blocks.segments.properties]
      cache_duration = 'none'

[[blocks]]
  type = 'rprompt'
  overflow = 'hidden'

  [[blocks.segments]]
    template = ' '
    foreground = 'p:green'
    background = 'transparent'
    type = 'node'
    style = 'plain'

    [blocks.segments.properties]
      cache_duration = 'none'
      display_mode = 'files'
      fetch_package_manager = false
      home_enabled = false

  [[blocks.segments]]
    template = ' '
    foreground = 'p:blue'
    background = 'transparent'
    type = 'go'
    style = 'plain'

    [blocks.segments.properties]
      cache_duration = 'none'
      fetch_version = false

  [[blocks.segments]]
    template = ' '
    foreground = 'p:yellow'
    background = 'transparent'
    type = 'python'
    style = 'plain'

    [blocks.segments.properties]
      cache_duration = 'none'
      display_mode = 'files'
      fetch_version = false
      fetch_virtual_env = false

  [[blocks.segments]]
    leading_diamond = ''
    trailing_diamond = ''
    template = ' <p:white></> {{ .FormattedMs }} '
    foreground = 'p:white'
    background = 'p:purple'
    type = 'executiontime'
    style = 'diamond'

    [blocks.segments.properties]
      cache_duration = 'none'
      style = 'austin'
      threshold = 500
```

I did find a half way of fixing it by instead using `leading_diamond = ''` but that has led to quite a few issues with font rendering on Wezterm (and maybe others), where it appears to be much shorter than the other characters in the prompt

**Reply by SirenGlitch** [reply]
How it looks on Kitty (what i want it to look like)

![image](https://github.com/user-attachments/assets/d5130957-8b94-47fa-8533-0962015582ec)

Versus on Wezterm. Interestingly, the height issue fixes itself when using a non-"Regular" font weight, but that's an issue for another repo
![image](https://github.com/user-attachments/assets/37256a8f-b234-4743-ae8e-ea8eaabfa1fa)
